src/cmds/fs/uniq.c

```diff
@@ -0,0 +1,185 @@
+/**
+ * @file
+ * @brief Filter  adjacent matching lines from FILE, writing to standard output.
+ *
+ * @date 02.07.19
+ * @author Nastya Nizharadze
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+#include <unistd.h>
+#include <errno.h>
+
+static int number_prefix;
+static int dup_only;
+static int nul_end;
+static int all_dup;
+
+static int max_length;
+static int dup_count;
+
+static void print_usage(void) {
+	printf(
+		"Usage: uniq [OPTION]... [FILE]...\n"
+		"Filter  adjacent matching lines from FILE, "
+		"writing to standard output.\n"
+		"\nOptions:\n\n"
+		"  -c\tprefix lines by the number of occurrences\n"
+		"  -d\tonly print duplicate lines, one for each group\n"
+		"  -D\tprint all duplicate lines\n"
+		"  -z\tline delimiter is NUL, not newline\n"
+		"  -h\tdisplay this help and exit\n"
+	);
+}
+
+static int read_line(char* line_buff, FILE *fp) {
+	if (NULL != fgets(line_buff, max_length + 3, fp)) {
+		int str_size = strlen(line_buff);
+		if (line_buff[str_size - 1] == '\n') {
+			if (1 == nul_end) {
+				line_buff[--str_size] = '\0';
+			}
+			else {
+				line_buff[str_size] = '\0';
+			}
+		}
+	}
+	else {
+		return -1;
+	}
+	return 0;
+}
+
+static void print_with_flags(char* line_buff) {
+	if (1 == number_prefix) {
+		if (1 == dup_only) {
+			if (1 != dup_count) {
+				printf("	%i %s", dup_count, line_buff);
+			}
+		}
+		else {
+			printf("	%i %s", dup_count, line_buff);
+		}
+	}
+	else {
+		if (1 == all_dup) {
+			if (1 != dup_count) {
+				for (int i = 0; i < dup_count; i++) {
+					printf("%s", line_buff);
+				}
+			}
+		}
+		else {
+			if (1 == dup_only) {
+				if (1 != dup_count) {
+					printf("%s", line_buff);
+				}
+			}
+			else {
+				printf("%s", line_buff);
+			}
+		}
+	}
+}
+
+int main(int argc, char **argv) {
+	int opt, i = 0;
+	char buff, *line_buff_curr, *line_buff_prev, *tmp;
+	FILE *fp;
+
+	number_prefix = 0;
+	dup_only = 0;
+	nul_end = 0;
+	all_dup = 0;
+	max_length = 0;
+	dup_count = 0;
+
+	if (argc < 2) {
+		print_usage();
+		return -EINVAL;
+	}
+
+	while (-1 != (opt = getopt(argc, argv, "cdDzh"))) {
+		switch (opt) {
+		case 'c':
+			number_prefix = 1;
+			break;
+		case 'd':
+			dup_only = 1;
+			break;
+		case 'D':
+			all_dup = 1;
+			break;
+		case 'z':
+			nul_end = 1;
+			break;
+		case '?':
+			printf(
+				"uniq: invalid option -- '%c'\n"
+				"Try 'uniq -h' for more information.\n",
+				optopt
+			);
+			return 0;
+		case 'h':
+			print_usage();
+			return 0;
+		default:
+			return -EINVAL;
+		}
+	}
+
+	if ((0 != number_prefix) && (0 != all_dup)) {
```

Comment:
This can be written like this:
`if (!number_prefix && !all_dup)`

Reply:
Done.